### PR TITLE
Vue component/SPA updates

### PR DESCRIPTION
**Pull request by xmas**
Migrate to making the Vue SPA as the default/only view renderer.

### CODE REVIEW

File: resources/js/components/OptOut/OptOut.vue [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+<p>Please choose a region below and click "opt-out". You will be redirected to Blizzard's secure authentication and asked to allow access for your Blizz ID and Battletag. This is the only information we use, and we do not store this information anywhere. We do not collect or store any personal information. After you have opted out, you may remove your authorization for this website by going to your Battle.net account under "Security &amp; Privacy" and managing your authorized applications.</p>
+<p>Once you have opted-out, data uploaded from replays in which you played a game will still be used to calculate data across the site, but your Battletag and Profile page will be removed.</p>
+
+<!--Need to add a confirm check here before allowing them to opt out-->
+<p>Please be aware that opting out also removes your data for use by others using our API tool.  So if your MMR is required for Amateur series, this will cause your data not to show up for them, and may make you ineligable for league play.<p>
```
Comment (Zemill) [inline resources/js/components/OptOut/OptOut.vue]: This is the only thing I saw missing from the opt out page. 

The user needs to accept the above statements before they can click the opt out button

File: routes/web.php [hunk 2]
```diff
@@ -65,14 +31,8 @@
 
 
 
-
+//Likely need to remove these and throw them as the API call.
+//Right now the API call just calls the HeroController @show method, but makes more sense to just use the API call to ask for discrete data 
+//similiar to how I am getting/returning data from view Global/Hero/test.blade.php
 Route::post('/get_heroes_stats_table_data', 'HeroController@getHeroStatsTableData')->name('get_heroes_stats_table_data');
```
Comment (Zemill) [inline routes/web.php]: //Likely need to remove these and throw them as the API call.
//Right now the API call just calls the HeroController @show method, but makes more sense to just use the API call to ask for discrete data 
//similiar to how I am getting/returning data from view Global/Hero/test.blade.php

File: routes/web.php [hunk 3]
```diff
@@ -14,45 +14,11 @@
...
-  Route::get('/Global/Hero/', 'HeroController@show');
-
+  Route::view('/', 'layouts.app');
 
-  */
-  Route::view('/Global/Leaderboard', 'Global/leaderboard');
-  Route::view('/Global/Hero/Talents', 'Global/Hero/talents');
+  //This route gives the correct object/data to use for the replacement code for getHeroStatsTableData.
+  //But the return is likely not what is expected so not changing it yet
   Route::view('/Global/Hero', 'Global/Hero/test');
```
Comment (Zemill) [inline routes/web.php]: //This route gives the correct object/data to use for the replacement code for getHeroStatsTableData.
  //But the return is likely not what is expected so not changing it yet

**Comment by lifeinabstract** [comment]
Data in the DataTable & SearchForm components was hard coded (by me when working on this originally). This needs to be separated out into the data for the overall components (GlobalLeaderboard, for example), so that DataTable & SearchForm can remain independent elements and be reused across several pages. Parts that need to be moved:
DataTable.vue
- fields data
- "dataurl" - pulled from mounted function and referenced by dataurl parameter inside DataTable.vue instead

SearchForm.vue
- form fields should be moved to their own components for each form field
- form fields - This may require more work, but my original idea was to have all form fields dynamically created/controlled by json, with a json file with name, type of input, options, etc. as of now, the structure doesn't allow this to be dynamic between pages using the SearchForm component
- "dataurl" - same as the Datatable.vue - move this to be dynamic; there is also a store commit in mounted that needs to use a dynamic URL instead

File: app/Http/Controllers/HeroController.php [hunk 4]
```diff
@@ -88,7 +88,7 @@ public function show(){
           "description" => "Major patches",
           "conditional_field" => "timeframe_type",
           "conditional_value" => "major",
-          "options" => $global->convertToFilter(Session::get('all_major_patch')),
+          // "options" => $global->convertToFilter(Session::get('all_major_patch')),
```
Comment (denizsokullu) [inline app/Http/Controllers/HeroController.php]: Hello! What's the reason why you are commenting these out and not removing them completely?

File: package.json [hunk 5]
```diff
@@ -5,35 +5,34 @@
         "development": "cross-env NODE_ENV=development node_modules/webpack/bin/webpack.js --progress --hide-modules --config=node_modules/laravel-mix/setup/webpack.config.js",
         "watch": "npm run development -- --watch",
         "watch-poll": "npm run watch -- --watch-poll",
-        "hot": "cross-env NODE_ENV=development node_modules/webpack-dev-server/bin/webpack-dev-server.js --inline --hot --config=node_modules/laravel-mix/setup/webpack.config.js",
+        "hot": "cross-env NODE_ENV=development node_modules/webpack-dev-server/bin/webpack-dev-server.js --inline --hot --https --config=node_modules/laravel-mix/setup/webpack.config.js --disableHostCheck=true",
```
Comment (denizsokullu) [inline package.json]: According to: https://webpack.js.org/configuration/dev-server/#devserverdisablehostcheck, it is true by default?

**Comment by xmas** [comment]
I need to chat w/ @Zemill on the options stuff. I had some issue making it work, but wasn’t sure of the final solution.

On the npm run config… Chrome on my Mac with the changes to .dev being a reserved domain was causing some issues… adding this fixed it and made HMR work correctly.

--
Rowan Christmas
rowanxmas@gmail.com
206.498.6689

File: package.json (hunk 5 above)
Comment (xmas) [reply package.json:8]: wasn't working... probably somewhere deep in the vue-cli setup got overridden...

**Pull request closed without merging**